delft/hydra-proxy.nix

```diff
@@ -1,71 +1,35 @@
 { config, lib, pkgs, ...}:
 
-let
-
-  hydraProxyConfig =
-    ''
-      TimeOut 900
-
-      <Proxy *>
-        Order deny,allow
-        Allow from all
-      </Proxy>
-
-      ProxyRequests     Off
-      ProxyPreserveHost On
-      ProxyPass         /apache-errors !
-      ProxyPass         /.well-known !
-      ErrorDocument 503 /apache-errors/503.html
-      ProxyPass         /       http://127.0.0.1:3000/ retry=5 disablereuse=on
-      ProxyPassReverse  /       http://127.0.0.1:3000/
-
-      <Location />
-        SetOutputFilter DEFLATE
-        BrowserMatch ^Mozilla/4\.0[678] no-gzip\
-        BrowserMatch \bMSI[E] !no-gzip !gzip-only-text/html
-        SetEnvIfNoCase Request_URI \.(?:gif|jpe?g|png|narinfo)$ no-gzip dont-vary
-        SetEnvIfNoCase Request_URI /api/ no-gzip dont-vary
-        SetEnvIfNoCase Request_URI /download/ no-gzip dont-vary
```

Comment:
I think we want to keep the no compression for download and nar since they are already compressed and nginx will just use CPU cycles. We should also probably disable proxy buffering for those

Reply:
The largest part of /download/ requests aren't already compressed. They're things like https://nixos.org/hydra/manual/ which is served straight out of https://hydra.nixos.org/build/196107287/download/1/hydra/ (for example)

The rest is getting MIME-sniffed properly by nginx to not get re-compressed.

```
$ curl -vvv -sH 'Accept-encoding: gzip' https://hydra.nixos.org/build/196107287/download/1/hydra/css/chrome.css -o/dev/null
< content-encoding: gzip

$ curl -vvv -sH 'Accept-encoding: gzip' https://hydra.nixos.org/build/246764727/download/1/nixos-system-x86_64-linux.tar.xz -o/dev/null
< content-type: application/x-xz
(no content-encoding)
```

Disabling proxy buffering is a micro-optimization which doesn't even do anything here because Hydra serves downloads by buffering them into memory already, which is terrible in all sorts of ways, I wish I hadn't looked into that to reply to this review comment :)

Reply:
Oh, and /nar/ isn't even a thing. (anymore?)

```
# zgrep /nar/ access-hydra.nixos.org.log.[23456].gz
[nothing]
```